### DISCUSSION TITLE
Unknown remote '192.168.1.3' ipsec status.

### DISCUSSION THREAD

**Discussion by NikolaevichDEV** [strongswan/strongswan#1520] [Q&A]
Good afternoon! Please tell me, in connections to the VPN server, through the "ipsec status" command, I see the IP address 192.168.1.3, instead of the user's login, and the login is reproduced in the "remote EAP identity". When you turn off the VPN with all devices with the “frog” login, the connection below is still good. What connection is this, I can not understand. Details below, information changed.

ikev2-vpn[16]: ESTABLISHED 4 hours ago, 62.217.XXX.XX[my_domain.ru]...79.139.160.199[192.168.1.3]
ikev2-vpn[16]: Remote EAP identity: frog
ikev2-vpn[16]: IKEv2 SPIs: 68ea5f1dawe6a2d5_i 77fc168re523742a9_r*, rekeying disabled
ikev2-vpn[16]: IKE proposal: AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_1536
ikev2-vpn{7}: INSTALLED, TUNNEL, reqid 6, ESP in UDP SPIs: c186129e_i 0714m8dd_o
ikev2-vpn{7}: AES_CBC_256/HMAC_SHA2_256_128, 1286991 bytes_i (10506 pkts, 18173s ago), 102526496 bytes_o (83148 pkts, 17662s ago), rekeying disabled
ikev2-vpn{7}: 0.0.0.0/0 === 10.250.0.4/33

Information about one of the connections through the swanctl package is also displayed below. It is active with the user who is already disconnected from the VPN. The username is stored in remote-eap-id. How is this possible and what kind of connection is this?

ikev2-vpn: #16, ESTABLISHED, IKEv2, 68ea5f1daae6a2d5_i 77ftr168ca54742a9_r*
  local  'my_domain.ru' @ 62.217.XXX.XX[4500]
  remote '192.168.1.3' @ 79.139.160.199[1821] EAP: 'frog' [10.250.0.4]
  AES_CBC-256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_1536
  established 19607s ago
  ikev2-vpn: #7, reqid 6, INSTALLED, TUNNEL-in-UDP, ESP:AES_CBC-256/HMAC_SHA2_25                                                                                                                                                             6_128
    installed 20218s ago
    in  c186129d, 1286991 bytes, 10506 packets, 20057s ago
    out 0714a8dd, 102526496 bytes, 83148 packets, 19546s ago
    local  0.0.0.0/0
    remote 10.250.0.4/33

**Comment by tobiasbrunner** [comment]
> I see the IP address 192.168.1.3, instead of the user's login,

That's the IKE identity (IDi) the client sends when it connects. When using EAP authentication that could basically be anything. Windows clients e.g. use their physical IP address, which is probably what you see here.

> and the login is reproduced in the "remote EAP identity".

Yes, that's the identity exchanged via EAP.

> When you turn off the VPN with all devices with the “frog” login

What does that mean? "turn off" how? Why not match the EAP identity (or both)?

**Reply by NikolaevichDEV** [reply]
Thanks for the answer. 
I connect to VPN server using the login and password from the ipsec.secrets file.
When I connected, its connection is available via ipsec status or swanctl -l. In the connections we see my IP and login.

If I disconnect from the server with my login, I keep seeing an entry in the connections where remote '192.168.1.3' and Remote EAP = my login. I have disabled the VPN, but I don't understand what the connection is from my account.

**Reply by tobiasbrunner** [reply]
Sorry, I don't understand what you are saying. Please provide more details (commands, logs etc.) for the steps you are actually performing.

**Reply by NikolaevichDEV** [reply]
Sorry, I'll try to explain.

**First case**:
When I connect to the VPN, I see this connection:

ikev2-vpn: #74, ESTABLISHED, IKEv2, 365355ebt934fad9d_i fe3b2f64f3181e27_r*
  local  'my_domain.ru' @ 62.217.1XX.XX[4500]
  remote 'efanov' @ 91.193.177.35[64910] [10.250.0.2]
  AES_CBC-256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
  established 11218s ago
  ikev2-vpn: #77, reqid 19, INSTALLED, TUNNEL-in-UDP, ESP:AES_CBC-256/HMAC_SHA2_                                                                                                                                                             256_128/MODP_2048
    installed 4222s ago
    in  ca2d8a12, 4494559 bytes, 29193 packets,    20s ago
    out e8fa50ae, 240182033 bytes, 196629 packets,    20s ago
    local  0.0.0.0/0
    remote 10.250.0.2/32

After some time, a second connection appears with my login "efanov":

ikev2-vpn: #16, ESTABLISHED, IKEv2, 68ea5f1dardwe6af2d5_i 77fc168ca54742a9_r*
  local  'my_domain.ru' @ 62.217.1XX.XX[4500]
  remote '192.168.1.3' @ 79.139.160.199[1821] EAP: 'efanov' [10.250.0.4]
  AES_CBC-256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_1536
  established 31542s ago
  ikev2-vpn: #7, reqid 6, INSTALLED, TUNNEL-in-UDP, ESP:AES_CBC-256/HMAC_SHA2_256_128
    installed 32153s ago
    in  c186129d, 1286991 bytes, 10506 packets, 31992s ago
    out 0714a8dd, 102526496 bytes, 83148 packets, 31481s ago
    local  0.0.0.0/0
    remote 10.250.0.4/32

I don't understand what is it. In the previous connection I saw my IP address (91.193.177.35), in this connection the IP address is 192.168.1.3. If I disable the VPN, the second connection is still visible via swanctl -l. 


**Second case**:

Now I see two connections in swanctl -l -P:

  ikev2-vpn {
    uniqueid = 107
    version = 2
    state = ESTABLISHED
    local-host = 62.217.1XX.XX
    local-port = 4500
    local-id = my_domain.ru
    remote-host = 178.176.78.88
    remote-port = 11041
    remote-id = 100.116.45.44
    remote-eap-id = efanov
    initiator-spi = 6862d6ed9ef7989c2
    responder-spi = 2a070f147041e0d8
    nat-remote = yes
    nat-any = yes
    encr-alg = AES_CBC
    encr-keysize = 256
    integ-alg = HMAC_SHA2_256_128
    prf-alg = PRF_HMAC_SHA2_256
    dh-group = MODP_1536
    established = 215
    remote-vips = [
      10.250.0.8
    ]

and

  ikev2-vpn {
    uniqueid = 101
    version = 2
    state = ESTABLISHED
    local-host = 62.217.1XX.XX
    local-port = 4500
    local-id = my_domain.ru
    remote-host = 178.176.78.88
    remote-port = 29489
    remote-id = 100.116.45.44
    remote-eap-id = efanov
    initiator-spi = 3ab19d92fsbed36a4
    responder-spi = 9b2f7bfde94bd1886
    nat-remote = yes
    nat-any = yes
    encr-alg = AES_CBC
    encr-keysize = 256
    integ-alg = HMAC_SHA2_256_128
    prf-alg = PRF_HMAC_SHA2_256
    dh-group = MODP_1536
    established = 3675
    remote-vips = [
      10.250.0.6
    ]

Me and my girlfriend have access to my login "efanov". 
If I disconnected the VPN connection, why do the logs show two connections through my login?
If my girlfriend is connected, there should be one connection, but now there are two.
When I connect to the VPN and enable/disable Wi-Fi, there might be a new entry in swanctl -l connections, which then doesn't disappear if I disable the VPN. This is my guess.

If I explained poorly, I'm sorry, anyway, your resource was useful to me! Thank you!

**Reply by NikolaevichDEV** [reply]
Sorry, I have one more question.
Is it possible to restrict connection access from one login to only one IP?
I want to deny access to other users if one is already connected to the VPN with the same login.

**Reply by tobiasbrunner** [reply]
> I don't understand what is it. In the previous connection I saw my IP address (91.193.177.35), in this connection the IP address is 192.168.1.3.

That's not the IP address, you are misunderstanding that output.

    remote 'efanov' @ 91.193.177.35[64910] [10.250.0.2]

This means that a client with the IP `91.193.177.35` and port `64910` connected using IKE identity `efanov`. It used the same identity as EAP identity to authenticate and was assigned the virtual IP address `10.250.0.2`.

     remote '192.168.1.3' @ 79.139.160.199[1821] EAP: 'efanov' [10.250.0.4]

This means that a client with IP `79.139.160.199` and port `1821` connected using the IKE identity `192.168.1.3`.  It then used a different identity during EAP, `efanov`, and was assigned the virtual IP address `10.250.0.4`.

So there are two clients connected using the same login credentials. The first is probably a Linux client, the second a Windows client (as mentioned, the Windows IKEv2 client sends its physical IP address as IKE identity before it authenticates via EAP using the actual username).

> If I disable the VPN, the second connection is still visible via swanctl -l.

What do you mean with "disable the VPN"? On the server somehow? Or on one of the clients? Why should the latter affect the second connection?

Clients can obviously create new connections as they need them, so there could definitely be multiple connections (even from the same client if it didn't delete the previous SA properly, e.g. if it wasn't able to because there was no network connectivity). These SAs will eventually get removed when there is an attempt to rekey them and the client won't respond (you can also configure DPD to force this removal earlier, but don't configure it too aggressive if clients are mobile and could be without network connectivity for a while).

> Is it possible to restrict connection access from one login to only one IP?

Have a look at the `unique` option.
